packages/narwallets/src/lib/narwallets.ts

```diff
@@ -0,0 +1,371 @@
+import { isMobile } from "is-mobile";
+import {
+  WalletModuleFactory,
+  WalletBehaviourFactory,
+  InjectedWallet,
+  Action,
+  Transaction,
+  FunctionCallAction,
+  Optional,
+  Account,
+} from "@near-wallet-selector/core";
+import { waitFor } from "@near-wallet-selector/core";
+import type {
+  InjectedNarwallets,
+  SignAndSendTransactionParams,
+  SignAndSendTransactionResponse,
+} from "./injected-narwallets";
+import { fileURLToPath } from "url";
+import { resolve } from "path";
+import { providers } from "near-api-js";
+import { CLIENT_RENEG_LIMIT } from "tls";
+
+// declare global {
+//   interface Window {
+//     narwallets: InjectedNarwallets | undefined;
+//   }
+// }
+
+export interface NarwalletsParams {
+  iconUrl?: string;
+}
+
+interface NarwalletsState {
+  wallet: InjectedNarwallets;
+}
+
+interface PendingPromises {
+  id: number;
+  code: string;
+  resolve: (value: any) => void;
+  reject: (reason?: any) => void;
+  timeout?: NodeJS.Timeout;
+}
+
+let id = 0;
+const pendingPromises: Array<PendingPromises> = [];
+
+const isInstalled = (): Promise<boolean> => {
+  const code = "is-installed";
+  return sendToNarwallets(code, true);
+};
+
+const isSignedIn = (): Promise<boolean> => {
+  const code = "is-signed-in";
+  return sendToNarwallets(code, true);
+};
+
+const getAccountId = (): Promise<string> => {
+  const code = "get-account-id";
+  return sendToNarwallets(code, true);
+};
+
+interface SignAndSendResponseError {
+  error: string;
+}
+
+const callSignAndSendTransaction = (
+  params: SignAndSendTransactionParams
+): Promise<providers.FinalExecutionOutcome | SignAndSendResponseError> => {
+  const code = "sign-and-send-transaction";
+  return sendToNarwallets(code, false, params);
+};
+
+const callSignAndSendTransactions = (
+  params: Array<SignAndSendTransactionParams>
+): Promise<
+  Array<providers.FinalExecutionOutcome> | SignAndSendResponseError
+> => {
+  const code = "sign-and-send-transactions";
+  return sendToNarwallets(code, false, params);
+};
+
+const sendToNarwallets = (
+  code: string,
+  withTimeout = false,
+  params?: any
+): Promise<any> => {
+  const promise = new Promise<any>((resolve, reject) => {
+    id++;
+    let promiseTimeout;
+    if (withTimeout) {
+      promiseTimeout = setTimeout(() => {
+        return reject(Error("timeout"));
+      }, 1000);
+    }
```

Comment:
This is always throwing an error and the Narwallets option is not showing up in the modal so I can not test the functionality of this wallet.
Error in cosole: **Error: timeout at eval (narwallets.ts?4493:93:23)**